lib/MTT/Test/Build.pm

```diff
@@ -416,6 +416,12 @@ sub _do_build {
     } elsif ($config->{srcdir}) {
         $ret = MTT::Module::Run("MTT::Test::Build::$config->{module}",
                                 "Build", $ini, $mpi_install, $config);
+        if ($ret->{noop_test_bin_dir}) {
+            # At Build step 'srcdir' is used to point to location of binaries.
+            # For Noop module it is possible to redefine it with external test
+            # location.
+            $config->{srcdir} = $ret->{noop_test_bin_dir};
```

Comment:
Thanks for this new PR.

But this still has two of the same problems I mentioned in the last PR:

1. You're using a module-specific value here in the core `Build.pm`.  This is an abstraction violation --
 `Build.pm` should be completely agnostic of all Build modules.
1. You're overriding `srcdir` with a value that is *not* the source directory.  This is very confusing -- i.e., a value named `$config->{srcdir}` *isn't* the source directory.  If you want to allow the test executables to be in a different directory than the source dir, then make a new `$config->{bindir}` field (that presumably defaults to the same value of `$config->{srcdir}`).  Use `$config->{bindir}` as appropriate here in `Build.pm` (and/or elsewhere -- perhaps `Run.pm` will need to use it too...?  You'll need to check).  Then `Build/Noop.pm` can just override `$config->{bindir}`.